client/src/plugins/usage-statistics/UsageStatistics.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import debug from 'debug';
+
+import { PureComponent } from 'react';
+
+import Flags, { ET_ENDPOINT, DISABLE_REMOTE_INTERACTION } from '../../util/Flags';
+import Metadata from '../../util/Metadata';
+
+import eventHandlers from './event-handlers';
+
+const log = debug('UsageStatistics');
+
+const PRIVACY_PREFERENCES_CONFIG_KEY = 'editor.privacyPreferences';
+const USAGE_STATISTICS_CONFIG_KEY = 'ENABLE_USAGE_STATISTICS';
+const EDITOR_ID_CONFIG_KEY = 'editor.id';
+
+const PRODUCT_MODELER = 'Modeler';
+const PRODUCT_EDITION_COMMUNITY = 'Community';
+
+const FETCH_METHOD = 'POST';
+const FETCH_HEADERS = { 'Accept': 'application/json', 'Content-Type': 'application/json' };
+
+// ET endpoint is set to Travis as an env variable, passed to client via WebPack DefinePlugin
```

Comment:
I'd suggest to remove this comment as if we ever change build flow the comment gets outdated. I doubt anybody would remember this.

```suggestion
```

Reply:
🐵

Reply:
I think it's good to keep this. We have a similar comment in [ErrorTracking](https://github.com/camunda/camunda-modeler/blob/af5f417a58cafcab5567cb71a6b36de7b655a985/client/src/plugins/error-tracking/ErrorTracking.js#L29) as well.

Rationale: It's not common to have `process.env.something` in browser javascript code. We should have comments like this to reduce confusions. If we change our building techniques, it'd still be good to have this comment there because it signals us what's missing and why certain feature are not working anymore.

Reply:
Comment is OK as long as it's not tightly coupled with build details. What I don't like about the comment is that it mentions our CI provider and build tool. What if we switch to CloudCI and Rollup?

Alternative:

```suggestion
// Replaced during build with exact endpoint
```

Reply:
You're right about the part where we should not mention about Travis.

I updated the messages in this plugin and ErrorTracking plugin via https://github.com/camunda/camunda-modeler/pull/1840/commits/cfea31d2c27f81bf7058a3f88b7481a3c3b40626

Reply:
I'm still keeping the Webpack stuff because if we get rid of webpack -> these plugins would not function -> they should not function -> we want to see them fail to fix them.

Reply:
OK, but we still mention Webpack DefinePlugin in the new comment. Why?

Reply:
I updated my comment.

Reply:
` we want to see them fail to fix them. -> it's good to see this functionality depends on WebPack DefinePlugin in the comments.`

Reply:
So let's say we started using some other thing than Webpack and as a result these plugins fail, it's good to see a comment about Webpack dependency to figure out why they failed.